tests/testthat/test-datatype.R

```diff
@@ -42,42 +42,32 @@ test_that("POSIXct data conversion", {
     pl$lit("2022-01-01")$str$strptime(pl$Datetime(), "%F")$to_r(),
     as.POSIXct("2022-01-01")
   )
-  # TODO: infer timezone from string, change the arugment name from `tz`
+
+  # Note: do not set the "TZ" envvar here as it doesn't work consistently
+  # https://github.com/pola-rs/r-polars/issues/1188
+  expect_identical(
+    pl$lit("2022-01-01")$str$strptime(pl$Datetime(time_zone = "America/New_York"), "%F")$to_r(),
+    as.POSIXct("2022-01-01", tz = "America/New_York")
+  )
   expect_true(
     as_polars_series("2022-01-01 UTC")$str$strptime(pl$Datetime(time_zone = "UTC"), "%F %Z")$eq(
       as_polars_series(as.POSIXct("2022-01-01", tz = "UTC"))
     )$to_r()
   )
 
-  withr::with_envvar(
-    new = c(TZ = "America/New_York"),
-    {
-      expect_identical(
-        pl$lit("2022-01-01")$str$strptime(pl$Datetime(), "%F")$to_r(),
-        as.POSIXct("2022-01-01")
-      )
-      # TODO: infer timezone from string, change the arugment name from `tz`
-      expect_true(
-        as_polars_series("2022-01-01 UTC")$str$strptime(pl$Datetime(time_zone = "UTC"), "%F %Z")$eq(
-          as_polars_series(as.POSIXct("2022-01-01", tz = "UTC"))
-        )$to_r()
-      )
-
-      non_existent_time_chr = "2020-03-08 02:00:00"
-      ambiguous_time_chr = "2020-11-01 01:00:00"
-      expect_identical(
-        pl$lit(as.POSIXct(non_existent_time_chr))$to_r(),
-        as.POSIXct(non_existent_time_chr)
-      )
-      expect_grepl_error(
-        pl$lit(non_existent_time_chr)$str$strptime(pl$Datetime(), "%F %T")$to_r(),
-        "non-existent"
-      )
-      expect_grepl_error(
-        pl$lit(ambiguous_time_chr)$str$strptime(pl$Datetime(), "%F %T")$to_r(),
-        "ambiguous"
-      )
-    }
+  non_existent_time_chr = "2020-03-08 02:00:00"
+  ambiguous_time_chr = "2020-11-01 01:00:00"
+  expect_identical(
+    pl$lit(as.POSIXct(non_existent_time_chr))$to_r(),
+    as.POSIXct(non_existent_time_chr)
+  )
+  expect_grepl_error(
+    pl$lit(non_existent_time_chr)$str$strptime(pl$Datetime(time_zone = "America/New_York"), "%F %T")$to_r(),
+    "non-existent"
+  )
+  expect_grepl_error(
+    pl$lit(ambiguous_time_chr)$str$strptime(pl$Datetime(time_zone = "America/New_York"), "%F %T")$to_r(),
+    "ambiguous"
```

Comment:
What should be tested here is the event of an error in converting naive time to R, which has been replaced by another test that tests for errors when converting datetime with time zone to R.

Reply:
No, the point of both these test is to ensure that invalid timestamps are caught. The only difference to the previous test is the explicit parameter for `time_zone` in the call to polars. This has nothing to do with naive time. Both parsed values are valid times without knowledge of time_zone, they are only invalid if the time zone is `America/New_York`

Reply:
> No, the point of both these test is to ensure that invalid timestamps are caught. The only difference to the previous test is the explicit parameter for `time_zone` in the call to polars. This has nothing to do with naive time. Both parsed values are valid times without knowledge of time_zone, they are only invalid if the time zone is `America/New_York`

I believe I added these tests, but you claim to know more about these tests than I?
Please check #878.